packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -225,6 +229,25 @@ class SliverConstraints extends Constraints {
   /// contents depends on the [growthDirection].
   final double scrollOffset;
 
+  /// The scroll distance that has been consumed by all [Sliver]s that came
+  /// before this [Sliver].
+  ///
+  /// [Sliver]s are often created lazily as layout occurs, e.g., [SliverList].
```

Comment:
currently slivers themselves are never _created_ lazily. It's the contents that are created lazily.